controllers/seedgen_controller.go

```diff
@@ -0,0 +1,644 @@
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/tools/record"
+	"k8s.io/client-go/util/retry"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	seedgenv1alpha1 "github.com/openshift-kni/lifecycle-agent/api/seedgenerator/v1alpha1"
+	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/clusterinfo"
+	"github.com/openshift-kni/lifecycle-agent/ibu-imager/ops"
+	"github.com/openshift-kni/lifecycle-agent/internal/common"
+
+	commonUtils "github.com/openshift-kni/lifecycle-agent/utils"
+)
+
+// SeedGeneratorReconciler reconciles a SeedGenerator object
+type SeedGeneratorReconciler struct {
+	client.Client
+	Log      logr.Logger
+	Scheme   *runtime.Scheme
+	Recorder record.EventRecorder
+	Executor ops.Execute
+}
+
+var (
+	clusterName            string
+	lcaImage               string
+	seedgenWorkingDir      = "/tmp/ibu-seedgen-orch"
+	seedgenAuthFile        = filepath.Join(seedgenWorkingDir, "auth.json")
+	storedSeedgenCR        = filepath.Join(seedgenWorkingDir, "seedgen-cr.json")
+	storedSeedgenSecret    = filepath.Join(seedgenWorkingDir, "seedgen-secret.json")
+	storedManagedClusterCR = filepath.Join(seedgenWorkingDir, "managedcluster.json")
+)
+
+//+kubebuilder:rbac:groups=lca.openshift.io,resources=seedgenerators,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups=lca.openshift.io,resources=seedgenerators/status,verbs=get;update;patch
+//+kubebuilder:rbac:groups=lca.openshift.io,resources=seedgenerators/finalizers,verbs=update
+//+kubebuilder:rbac:groups="",resources=configmaps,verbs=get;list;watch;create;update;patch;delete
+//+kubebuilder:rbac:groups="",resources=events,verbs=create;patch
+//+kubebuilder:rbac:groups="",resources=secrets,verbs=get;list;watch;delete
+//+kubebuilder:rbac:groups="",resources=namespaces,verbs=get;list;watch;delete
+//+kubebuilder:rbac:groups="",resources=pods,verbs=get;list;watch
+//+kubebuilder:rbac:groups=config.openshift.io,resources=clusterversions,verbs=get;list;watch
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterroles,verbs=delete
+//+kubebuilder:rbac:groups=rbac.authorization.k8s.io,resources=clusterrolebindings,verbs=delete
+//+kubebuilder:rbac:groups=apiextensions.k8s.io,resources=customresourcedefinitions,verbs=delete
+//+kubebuilder:rbac:groups=cluster.open-cluster-management.io,resources=managedclusters,verbs=get;list;watch;delete
+
+func (r *SeedGeneratorReconciler) stringToFile(filename, str string) error {
```

Comment:
Nit, this function doesn't use anything from SeedGeneratorReconciler?

Reply:
No, I mostly just grouped the various new bits I was adding to the reconciler, since that's where I was using them. Just a local utility function, but could easily move it to the utils module